test/fuse/linux/setstat_test.cc

```diff
@@ -0,0 +1,358 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/fuse.h>
+#include <sys/stat.h>
+#include <sys/statfs.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <utime.h>
+
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "test/fuse/linux/fuse_base.h"
+#include "test/util/fs_util.h"
+#include "test/util/fuse_util.h"
+#include "test/util/test_util.h"
+
+namespace gvisor {
+namespace testing {
+
+namespace {
+
+class SetStatTest : public FuseTest {
+ public:
+  SetStatTest()
+      : test_dir_("testdir"),
+        test_file_("testfile"),
+        test_dir_mode_(S_IFDIR | S_IRUSR | S_IWUSR | S_IXUSR),
+        test_file_mode_(S_IFREG | S_IRUSR | S_IWUSR | S_IXUSR) {}
+
+  void SetUp() override {
+    FuseTest::SetUp();
+    test_dir_path_ = JoinPath(mount_point_.path(), test_dir_);
+    test_file_path_ = JoinPath(mount_point_.path(), test_file_);
+  }
+  void TearDown() override {
+    EXPECT_EQ(GetServerNumUnsentResponses(), 0);
+    UnmountFuse();
+  }
+
+ protected:
+  PosixErrorOr<FileDescriptor> OpenPath(const std::string &path) {
```

Comment:
Will other integration tests call this function as well?

Reply:
I use this in `stat_test.cc` too though with different open flags. I tried to move this utility function to another library, but it turns out it depends on the functions of `fuse_base`. If we create another library and inherit from the new class, then all of the previous tests should be changed according to this change. Currently only this `OpenPath` function is used across the tests. Compared with the effort of creating a new library, I'll keep this local.

Reply:
`OpenPath` is added in #9.